### DISCUSSION TITLE
Error - rownames (genes) of individual RNA objects are not equivalent.

### DISCUSSION THREAD

**Discussion by sylestiel** [GreenleafLab/ArchR#2058] [General]
Hi @rcorces ,
Trying to integrate 3 multiome datasets. Having problems with running the import10xFeatureMatrix command:
 seRNA <- import10xFeatureMatrix(
    input = c("filePATH/rheMac10_LBN/628/outs/filtered_feature_bc_matrix.h5", "filePATH/rheMac10_LBN/623/outs/filtered_feature_bc_matrix.h5","filePATH/rheMac10_LBN/627/outs/filtered_feature_bc_matrix.h5"),names = c("L105", "L108", "L129"),strictMatch = TRUE
)
Importing Feature Matrix 1 of 3
Importing Feature Matrix 2 of 3
Importing Feature Matrix 3 of 3
Re-ordering RNA matricies for consistency.
Merging individual RNA objects...

Merging L108
### Error in import10xFeatureMatrix(input = c("/Volumes/BKUP2/rheMac10_LBN/628/outs/filtered_feature_bc_matrix.h5",  : 
  Error - rownames (genes) of individual RNA objects are not equivalent.

[ArchR-createArrows-1649b689979d6-Date-2023-11-21_Time-10-19-22.383064.log](https://github.com/GreenleafLab/ArchR/files/13428720/ArchR-createArrows-1649b689979d6-Date-2023-11-21_Time-10-19-22.383064.log).

Any suggestion would be much appreciated!
Thanks!

**Comment by sylestiel** [comment]
Hi@rcorces,

Still having issues with import10xFeatureMatrix() function when trying to load multiple files. Error persists. Can you point me towards a possible fix?

Thanks!

**Reply by liangliangjin** [reply]
Here's how I do it: If each database gene list is different, I loop through all files, extract the common gene list, and finally use "cbind" and "SummarizedExperiment" command to create a seRNA object.

**Reply by sylestiel** [reply]
Thanks @liangliangjin,
Do you mind sharing the actually scripts for the loop and cbind steps?
I changed from Seurat 5 to Seurat 4.3.0 and got the import10XFeatureMatrix() to work without a problem but then the next step which is addGeneExpressionMatrix() failed. Clearly, ArchR is not compatrible with Seurat5. However, I do not know how to address the addGeneExpressionMatrix() step failure. 
I do not have issues processing a single multiome dataset. The issue is with processing multiple multiome datasets. Looking forward to hearing back from you with a solution to my problems.

Thanks!

**Reply by liangliangjin** [reply]
@sylestiel Sure.
Because R package dependencies are complex, my versions are still ArchR 1.0.2 and Seurat 4.2.1.
Here is the code for combining the six samples, all you have to do is change the organ and the detailed file directory:

organ<-c("sample1","sample2","sample3","sample4","sample5","sample6")

for(i in 1:length(organ)){
assign(paste0("seRNA_",organ[i]),import10xFeatureMatrix(input = paste0("/home/",organ[i],"/filtered_feature_bc_matrix.h5"),names = organ[i]))##Read the .h5 data of all samples
}

seRNA_genelist<-rownames(get(paste0("seRNA_",organ[1])));
for(i in 2:length(organ)){
seRNA_genelist<-intersect(seRNA_genelist,rownames(get(paste0("seRNA_",organ[i]))))
}

b<-get(paste0("seRNA_",organ[1]))
seRNA<-assays(b[seRNA_genelist,])[[1]]
for(i in 2:length(organ)){
b<-get(paste0("seRNA_",organ[i]))
seRNA<-cbind(seRNA,assays(b[seRNA_genelist,])[[1]])
}

seRNA2<-SummarizedExperiment(assays=list(counts=seRNA), rowRanges= rowRanges(seRNA_sample1[seRNA_genelist,]))#Create a Summarized object, containing information such as chromosomes, rowRanges can just take any one of them, because all of them are the same after the intersection

proj <- addGeneExpressionMatrix(input = proj, seRNA = seRNA2)

**Reply by sylestiel** [reply]
Hi@liangliangjin 

The error is still present right after the step
for(i in 1:length(organ)){
assign(paste0("seRNA_",organ[i]),import10xFeatureMatrix(input = paste0("/home/",organ[i],"/filtered_feature_bc_matrix.h5"),names = organ[i]))##Read the .h5 data of all samples
}
Importing Feature Matrix 1 of 3
Importing Feature Matrix 2 of 3
Importing Feature Matrix 3 of 3
Re-ordering RNA matricies for consistency.
Merging individual RNA objects...

Merging NA
Error in import10xFeatureMatrix(input = paste0(input), names = samples[i]) : 
  Error - rownames (genes) of individual RNA objects are not equivalent.



How would I modify the script if I call in each of the sample seRNA is input separately (seRNA.1, seRNA.2, seRNA.3). import10XFeatureMatrix cannot be run for multiple samples without it giving the error.
